relayer/middleware/missedVaasV3/worker.ts

```diff
@@ -0,0 +1,149 @@
+import Redis, { Cluster } from "ioredis";
+import { Registry } from "prom-client";
+import { ChainId } from "@certusone/wormhole-sdk";
+import { createPool, Pool } from "generic-pool";
+import { Logger } from "winston";
+
+import { defaultWormholeRpcs, RelayerApp } from "../../application";
+import { mapConcurrent, sleep } from "../../utils";
+import { RedisConnectionOpts } from "../../storage/redis-storage";
+import { initMetrics, MissedVaaMetrics } from "./metrics";
+import { updateMetrics } from "./helpers";
+import {
+  runMissedVaaCheck,
+  refreshSeenSequences,
+  registerEventListeners,
+} from "./check";
+
+export interface MissedVaaOpts extends RedisConnectionOpts {
+  registry?: Registry;
+  logger?: Logger;
+  wormholeRpcs?: string[];
+  // How many "source" chains will be scanned for missed VAAs concurrently.
+  concurrency?: number;
+  // Interval at which the worker will check for missed VAAs.
+  checkInterval?: number;
+  // Times a VAA will be tried to be fetched when it's found missing
+  fetchVaaRetries?: number;
+  // Max concurrency used for fetching VAAs from wormscan.
+  vaasFetchConcurrency?: number;
+  /**
+   * "storagePrefix" is the prefix used by the storage (currently redis-storage) to
+   * store workflows. See RedisStorage.getPrefix
+   *
+   * This is generating a dependency with the storage implementation, which is not ideal.
+   * To solve this problem, we could add a new method to the storage interface to get seen sequences
+   * and pass it to the missed vaas middleware
+   *
+   * Untill that happens, we assume that if you pass in a storagePrefix property,
+   * then you are using redis-storage
+   */
+  storagePrefix?: string;
+  // The minimal sequence number the VAA worker will assume that should exist, by chain ID.
+  startingSequenceConfig?: Partial<Record<ChainId, bigint>>;
+  // If true the key will remove the existing sorted set containing seen keys and reindex all keys
+  // getting the from storage. This probably can be used in combination with startingSequenceConfig
+  // to force reprocessing of certain VAAs.
+  // WARN: skipped VAAs will be reprocessed.
+  forceSeenKeysReindex?: boolean;
+}
+
+export interface FilterIdentifier {
+  emitterChain: number;
+  emitterAddress: string;
+}
+
+let metrics: MissedVaaMetrics;
+
+export async function spawnMissedVaaWorker(
+  app: RelayerApp<any>,
+  opts: MissedVaaOpts,
+): Promise<void> {
+  opts.wormholeRpcs = opts.wormholeRpcs ?? defaultWormholeRpcs[app.env];
+  if (!metrics) {
+    metrics = opts.registry ? initMetrics(opts.registry) : {};
+  }
+
+  const redisPool = createRedisPool(opts);
+
+  if (!app.filters.length) {
+    opts.logger?.warn(
+      "Missed VAA Worker: No filters found, retrying in 100ms...",
+    );
+    setTimeout(() => {
+      spawnMissedVaaWorker(app, opts);
+    }, 100);
```

Comment:
Should we just fail instead of retrying indefinetly?

Reply:
`app.filters` are a very very basic configuration for any relayer app, so I expect them to always be present on any app.
The reason for this check is because you might want to start the missed-vaa worker before you setup the filters. So I just wait for some time until filters are already loaded into the app.
If a dev haven't setup filters in their application, they could see this log repeating infintely and understand what's going on... but, it's extremely unlikely that relayer will get to production without a filter in place